client/pages/auth/index.tsx

```diff
@@ -26,6 +27,24 @@ function ConnectButton() {
   };
   const Button = isConnected() ? ButtonGreen : ButtonGray;
 
+  useEffect(() => {
+    if (account) {
+      fetch('/api/auth', {
+        method: 'POST',
+        body: JSON.stringify({ walletAddress: account }),
+      })
+        .then((response) => {
+          console.log(response);
```

Comment:
In both the `.then` occurrences we'll have to check for errors and disconnect the wallet if so, that would imply something went wrong as we couldn't validate the address (wrong chain?) or create an account due to downtime.